clingon/clingon.py

```diff
@@ -135,20 +139,84 @@ def __init__(self, func):
             print('clize default parameters:',
                   self.reqargs + options.values() + (['*' + self.varargs] if self.varargs else []))
 
-    def override_defaults(self, options):
+    @staticmethod
+    def _check_booleans(options):
+        for k, v in iteritems(options):
+            if v is True:
+                raise ValueError("Default value for boolean option %r must be 'False'" % k)
+
+    def override_defaults_from_environ(self, options):
         prefix = self.variables.get('CLINGON_PREFIX')
         if prefix:
-            for k in options.keys():
+            for k in list(options):
                 default = os.environ.get(prefix + '_' + k.upper(), None)
                 if default is not None:
                     try:
                         default = eval(default, {}, {})
                     except (SyntaxError, NameError):
                         pass
                     options[k] = default
-        for k, v in iteritems(options):
-            if v is True:
-                raise ValueError("Default value for boolean option %r must be 'False'" % k)
+
+    def _get_dict_from_file(self, file, path=None):
+        if path:
+            file = os.path.join(path, file)
+        try:
+            file = glob.glob(file)[0]
+        except IndexError:
+            raise RuntimeError("Defaults parameter file %s not found" % file)
+        _, ext = os.path.splitext(file)
+        if ext == '.py':
+            defaults = {}
+            execfile(file, {}, defaults)
+        elif ext in ('.yml', '.yaml'):
+            import yaml
+            with open(file, 'r') as f:
+                defaults = yaml.load(f)
+        elif ext == '.json':
+            import json
+            with open(file, 'r') as f:
+                defaults = json.load(f)
+        else:
+            raise RuntimeError("Unknown defaults file format %s" % file)
+        return file, defaults
+
+    def override_defaults_from_file(self, options):
+        """
+        Search and read a configuration file to override defaults
+        search order is:
+        - hardcoded path in variable DEFAULTS_PATH if existing,
+        - local directory,
+        - ~/.clingon/,
```

Comment:
it's really minor, but I don't think anyone would want to store some stuff in `~/.clingon/` or `/etc/clingon/` because it would be global settings (and I think the use case is local settings.